test/f32-conv-hwc.cc

```diff
@@ -2,5047 +2,451 @@
+      for (size_t input_widths_ = (GetParam().padding_left ? 1 : 2); input_widths_ < GetParam().input_widths * 8;
+           input_widths_ += (GetParam().input_widths * 2 - 1)) {
+        auto tester = ConvHWCMicrokernelTester()
+          .kernel_size(GetParam().kernel_size)
+          .subsampling(GetParam().subsampling)
+          .input_channels(GetParam().input_channels)
+          .output_channels_tile(GetParam().output_channels_tile)
           .output_channels(output_channels)
-          .input_width(input_width)
-          .input_height(9)
-          .Test(xnn_f32_conv_hwc_ukernel_3x3s2p1c3x4__scalar_1x1, xnn_init_f32_minmax_scalar_params);
+          .input_width(input_widths_)
+          .input_height(9);
+        if (GetParam().padding_left == 1 && GetParam().padding_right == 1) {
+          tester.padding_width(GetParam().padding_right);
+        } else if (GetParam().padding_left == 0 && GetParam().padding_right == 1) {
+          tester.padding_right(GetParam().padding_right);
+        }
+        tester.padding_bottom(padding_bottoms).Test(GetParam().kernel);
       }
     }
   }
 }
 
-TEST(F32_CONV_HWC_3X3S2P1C3X4__SCALAR_1X1, padding_bottom) {
-  for (size_t padding_bottom = 0; padding_bottom <= 1; padding_bottom++) {
-    for (size_t output_channels = 1; output_channels < 8; output_channels += 3) {
-      for (size_t input_width = 1; input_width < 16; input_width += 3) {
-        ConvHWCMicrokernelTester()
-          .kernel_size(3)
-          .subsampling(2)
-          .padding_width(1)
-          .padding_bottom(padding_bottom)
-          .input_channels(3)
-          .output_channels_tile(4)
-          .output_channels(output_channels)
-          .input_width(input_width)
-          .input_height(9)
-          .Test(xnn_f32_conv_hwc_ukernel_3x3s2p1c3x4__scalar_1x1, xnn_init_f32_minmax_scalar_params);
+TEST_P(XnnTest, output_y_start)
+{
+  TEST_REQUIRES_ARCH_FLAGS(GetParam().arch_flags);
+  for (size_t output_y_starts = 1; output_y_starts <= 3; output_y_starts++) {
+    for (size_t output_channels = 1; output_channels < GetParam().output_channels_tile * 2;
+         output_channels += GetParam().output_channels_tile - 1) {
+      for (size_t input_widths_ = (GetParam().padding_left ? 1 : 2); input_widths_ < GetParam().input_widths * 8;
+           input_widths_ += (GetParam().input_widths * 2 - 1)) {
+        auto tester = ConvHWCMicrokernelTester()
+                          .kernel_size(GetParam().kernel_size)
+                          .subsampling(GetParam().subsampling)
+                          .input_channels(GetParam().input_channels)
+                          .output_channels_tile(GetParam().output_channels_tile)
+                          .output_channels(output_channels)
+                          .input_width(input_widths_)
+                          .input_height(9)
+                          .output_y_start(output_y_starts);
+        if (GetParam().padding_left == 1 && GetParam().padding_right == 1) {
```

Comment:
I suggest moving this logic into a new helper function of the tester, that takes the padding_left and padding_right params.